scarb/src/compiler/plugin/proc_macro/compilation.rs

```diff
@@ -37,7 +41,8 @@ impl SharedLibraryProvider for Package {
     }
 
     fn shared_lib_path(&self, config: &Config) -> Utf8PathBuf {
-        let lib_name = library_filename(self.id.name.to_string());
+        let lib_name = get_cargo_library_name(self).expect("could not resolve library name");
```

Comment:
should be context, not expect

Reply:
This method is `Utf8PathBuf` though, I don't think we can have errors there, and there's:
```rs
        let lib_name = lib_name
            .into_string()
            .expect("library name must be valid UTF-8");
```
just below

Reply:
Yes, it has been returning a plain `Utf8PathBuf` without `Result`, but it was also just doing simple operations on strings and not calling the Cargo metadata :D Now that it is calling the metadata (and thus is prone to erroring on user input), we need to wrap it into a result. The next line is simply converting `OsString` to `String`, which in that case should always succeed regardless of user input, so we can panic on that.

Reply:
Makes sense, addressed in [e44a218](https://github.com/software-mansion/scarb/pull/1605/commits/e44a218276b82ef0cfc720f3e5c304b44381310a) 👍🏻 

> We can discuss our strategy on when use panic vs result next week

Sounds good (although I think I got the overall idea 😄)